test/DataSet.test.js

```diff
@@ -1,44 +1,43 @@
 var assert = require('assert');
 var vis = require('../dist/vis');
-var moment = vis.moment;
 var DataSet = vis.DataSet;
 var Queue = vis.Queue;
 // TODO: test the source code immediately, but this is ES6
 
 var now = new Date();
 
 describe('DataSet', function () {
-  it('should work', function () {
-    // TODO: improve DataSet tests, split up in one test per function
+  var data = new DataSet({
+    type: {
+      start: 'Date',
+      end: 'Date'
+    }
+  });
 
-    var data = new DataSet({
-      type: {
-        start: 'Date',
-        end: 'Date'
-      }
-    });
+  // add single items with different date types
+  data.add({id: 1, content: 'Item 1', start: new Date(now.valueOf())});
+  data.add({id: 2, content: 'Item 2', start: now.toISOString()});
+  data.add([
+    //{id: 3, content: 'Item 3', start: moment(now)}, // TODO: moment fails, not the same instance
+    {id: 3, content: 'Item 3', start: now},
+    {id: 4, content: 'Item 4', start: '/Date(' + now.valueOf() + ')/'}
+  ]);
 
-    // add single items with different date types
-    data.add({id: 1, content: 'Item 1', start: new Date(now.valueOf())});
-    data.add({id: 2, content: 'Item 2', start: now.toISOString()});
-    data.add([
-      //{id: 3, content: 'Item 3', start: moment(now)}, // TODO: moment fails, not the same instance
-      {id: 3, content: 'Item 3', start: now},
-      {id: 4, content: 'Item 4', start: '/Date(' + now.valueOf() + ')/'}
-    ]);
+  var sort = function (a, b) {
+    return a.id > b.id;
+  };
+
+  it('can handle multiple data types', function () {
```

Comment:
With respect to usage of variable `data`, defined globally in the `describe`-block:

You are making a potentially dangerous assumption here, namely that the tests will *all* run, and will always run in the *same order*.

From the `mocha` command line usage:

```
-g, --grep <pattern>                    only run tests matching <pattern>
-f, --fgrep <string>                    only run tests containing <string>
```

I.e. it is possible to run specific tests, not all of them. In addition, a fix to `mocha` for optionally randomizing the test order is pending.

While this is not directly an issue with the current tests, it's something which time or a user can (and therefore will) break.

My suggestions here are:

- Wrap the initialization of `data` in an initialization function, call this in every `it`-block
- *and/or* initialize the data in a `beforeEach` block.

These can be combined. The goal in any case is to allow the `it`-blocks to run independently.